communication/mavlink_waypoint_handler.cpp

```diff
@@ -186,20 +164,20 @@ void Mavlink_waypoint_handler::receive_count(Mavlink_waypoint_handler* waypoint_
         if (waypoint_handler->waypoint_receiving_ == false)
```

Comment:
Yes that should be tested. I guess because of this line restarting sending of waypoint list takes a second click, then a timeout somewhere before we can resume  sending the waypoint list.